packages/react-scripts/scripts/eject.js

```diff
@@ -36,6 +37,31 @@ prompt(
     process.exit(1);
   }
 
+  // Make sure there are no dirty git status
+  const git = fs.existsSync(path.join(process.cwd(), '.git'));
```

Comment:
Hi @thejameskyle, can you explain what are the benefits of using that library? Is it because it returns promised result? Cause I was trying to avoid using third-party library as possible. If promise is necessary I could just use instantiate Promise.